### ISSUE TITLE
Mint: gesture triggers wrong action

### ISSUE THREAD

**Issue by ReaderGuy42** [jordansissel/xdotool#332]
I'm new to Linux and on the newest version of Mint.
The gesture that's supposed to go to the next workspace (Ctrl+shift+right) instead snaps the current window to the right and vice versa.
I can't figure out where Fusuma is getting the action related to that key combo from.

Any advice?
Thanks!

**Comment by jordansissel** [comment]
Hmm, can you explain a bit more? Specifically --

Do you mean that on your keyboard, when you press ctrl+shift+right, it behaves as you expect (go to next workspace), but when you use xdotool for this `xdotool key ctrl+shift+right` it does an unexpected behavior (snaps current window to the right)?

**Comment by ReaderGuy42** [comment]
Yes, that is correct, Ctrl+shift+right goes to the next workspace, but the Fusuma/xdotool is doing something else.

I'm using this as my config: https://github.com/iberianpig/fusuma/wiki/Linux-Mint-19.3-Cinnamon

Any ideas?

**Comment by jordansissel** [comment]
I haven’t tested this yet, but one hypothesis is that the keys xdotool
sends aren’t the same as your keyboard. X11 knows the difference between
left-control and right-control, and left-shift and right-shift… the
locations on the keyboard.

In this case, “ctrl” and “shift” key names (in xdotool) use the “left” ones
on the keyboard:
https://github.com/jordansissel/xdotool/blob/master/xdo_util.h#L15

It’s just a guess, but maybe try using the same specific keys you press on
the keyboard. So if you press left control, right shift, and right arrow,
try:

xdotool key Control_L+Shift_R+Right

Maybe?

Another option to investigate is to run the ‘xev’ program which can help
better understand what key symbols are being sent. Run ’xev’, and when the
window is focused (on top?) press your keys. The terminal output of xev
will show you what keys are sent.

**Comment by ReaderGuy42** [comment]
Thanks so much! In the meantime I've actually gotten it to run via libinput, earlier I'd tried via Fusuma. It's very likely the issue was rather my syntax, as I'm still extremely new to Linux.
I have all my swipes set as planned, now I'm still trying to figure out, why two finger pinch to zoom isn't working that well/reliably.

If you have the time, I'm curious about something: When moving to the next workspace (Ctrl+alt+right), the corresponding swipe works perfectly, as it just simulates the key presses. However, adding Shift is supposed to move the currently active window to the next workspace. Using the actual buttons that works fine. However when I hold shift and to the above swipe, it just moves over to the next workspace and the active window remains behind.

Any ideas why that would be?
Thanks :)
